Added HTTPHeaderDict to top level objects and updated the docs with i… (#2216)
urllib3/urllib3: docs/user-guide.rst

    >>> json.loads(r.data.decode('utf-8'))['headers']
    {'X-Something': 'value', ...}

> Or you can use the ``HTTPHeaderDict`` class to create a proper variable for storing header information.

Reviewer: ```suggestion
Or you can use the ``HTTPHeaderDict`` class to create multi-valued HTTP headers.
```

.. code-block:: pycon

    >>> from urllib3 import HTTPHeaderDict
    >>> Headers = HTTPHeaderDict()
    >>> Headers.add('X-Something', 'value')

>     >>> Headers.add('Y-Something', 'value')

Reviewer: Let's use a multi-value HTTP header here, for example Accept with both application/json and text/plain

    >>> print(Headers['X-Something'])
    value
    >>> for i in Headers.iteritems():
    ...     print( "%s: %s" % i)
    X-Something: value

>     Y-Something: value

Reviewer: You don't need those lines as we're already printing that below anyway

.. code-block:: pycon

    >>> logging.getLogger("urllib3").setLevel(logging.WARNING)

>     

Reviewer: Any reason to add this blank line? If not let's remove it
